### PR TITLE
Upgrade-to-vite

### PR DESCRIPTION
**Pull request by mayank1513**
- Replace react-scripts with vite for better build and live reload speed
  - better dx
  - modern tooling
- Set up vitest for unit testing
- Updated dependencies to latest
- Setup ESLint as per vite
- Set up prettier for consistent code formatting
- Refactor for better readability and maintainability 
- Fix lint issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1",
-    "typescript": "~5.2.2"
+    "@mui/icons-material": "^5.15.19",
+    "@mui/lab": "5.0.0-alpha.170",
+    "@mui/material": "^5.15.19",
+    "@mui/x-date-pickers": "^6.x",
+    "@types/node": "^20.14.2",
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
```
Comment (lucas-koehler) [inline package.json:15]: Can we move these to `devDependencies` please?

File: .gitignore [hunk 2]
```diff
@@ -31,6 +31,7 @@ bower_components
 
 # Compiled binary addons (http://nodejs.org/api/addons.html)
 build/Release
+dist
```
Comment (lucas-koehler) [inline .gitignore:34]: As the build is placed in the `build` directory, I believe we don't need this. Or am I missing something?

**Review by lucas-koehler — CHANGES_REQUESTED** [review]
Hi @mayank1513 ,

thanks for this contribution :heart: . Upgrading the seed app from CRA to ViteJS is certainly a good idea :+1: 

I have two remarks inline and noticed three further things:

- Linting no longer works. The whole `"eslintConfig"` property in the package.json is invalid as the referenced configs are part of CRA and, thus, no longer present. This should be removed. With this, there is no more lint config. I suppose it would make sense to use a vitejs default lint setup for typescript. Could you add this?
- For me the tests (via `npm run test`) cannot be executed because a dependency for `jsdom` is missing. We should add it as a dev dependency
- Optional: The build complains that the deprecated CJS (common js) build is used. We could adapt the package.json and tsconfig.json to use ESM.

File: package.json (hunk 1 above)
Comment (mayank1513) [reply package.json:15]: Sure

**Comment by mayank1513** [comment]
> Hi @mayank1513 ,
> 
> thanks for this contribution ❤️ . Upgrading the seed app from CRA to ViteJS is certainly a good idea 👍
> 
> I have two remarks inline and noticed three further things:
> 
>     * Linting no longer works. The whole `"eslintConfig"` property in the package.json is invalid as the referenced configs are part of CRA and, thus, no longer present. This should be removed. With this, there is no more lint config. I suppose it would make sense to use a vitejs default lint setup for typescript. Could you add this?
> 
>     * For me the tests (via `npm run test`) cannot be executed because a dependency for `jsdom` is missing. We should add it as a dev dependency
> 
>     * Optional: The build complains that the deprecated CJS (common js) build is used. We could adapt the package.json and tsconfig.json to use ESM.

Thank you for your observations. Probably I did not notice these as I was using yarn and pnpm locally. I have fixed the issues you observed.

**Comment by mayank1513** [comment]
Initially there was issue with even launching the Cypress test. I fixed
that. However the tests were not passing in the GitHub action - it was
throwing timeout error.

Can you help debug and fix this?

File: .vscode/extensions.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   // List of extensions which should be recommended for users of this workspace.
   "recommendations": [
     "esbenp.prettier-vscode",
-    "ms-vscode.vscode-typescript-tslint-plugin"
+    "mayank1513.trello-kanban-task-board"
```
[the remark this answers is not in the document]
Comment (mayank1513) [reply .vscode/extensions.json:8]: done

**Comment by mayank1513** [comment]
Cypress is failing because this element does not exist in the DOM - `[id="#/properties/recurrence"] > div`

**Comment by mayank1513** [comment]
Fixed Cypress tests...

**Review by lucas-koehler — APPROVED** [review]
Hi @mayank1513 , thank you very much for the updates and the whole rework! Changes LGTM now :sparkles: 

I added commits for three small changes:
- Fix one lint error in the cypress config
- Execute CI in pull requests and add linting to CI
- Regenerate the package-lock.json: We always do that for external/unknown contributors as a general security precaution.

**Pull request merged**